src/main/java/dao/H2Factory/H2CustomerDao.java

```diff
@@ -0,0 +1,105 @@
+package dao.H2Factory;
+
+import customerproductorder.models.Customer;
+import dao.CustomerDaoInterface;
+import dao.DaoException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class H2CustomerDao implements CustomerDaoInterface {
+    
+    private final ConnectionProvider connectionProvider;
+    
+    protected H2CustomerDao(ConnectionProvider connectionProvider) {
+        this.connectionProvider = connectionProvider;
+    }
+    
+    public void create(Customer newCustomer) throws DaoException {
+        try {
+            PreparedStatement st = connectionProvider.getConnection()
+                    .prepareStatement("insert into customerapplication"
+                            + ".customer(firstname,lastname,"
+                            + "address) values(?,?,?)");
+            st.setString(1, newCustomer.getFirstName());
+            st.setString(2, newCustomer.getName());
+            st.setString(3, newCustomer.getAddress());
+            st.executeUpdate();
+        } catch (SQLException e) {
+            throw new DaoException("Database connection error", e);
+        }
+    }
+    
+    public Customer get(int id) throws DaoException {
+        Customer gotCustomer = null;
+        try {
+            PreparedStatement st = connectionProvider.getConnection()
+                    .prepareStatement("select * from customerapplication"
+                            + ".customer where id=?");
+            st.setInt(1, id);
+            ResultSet rs = st.executeQuery();
+            gotCustomer = this.convertToCustomer(rs).get(0);
+        } catch (SQLException e) {
+            throw new DaoException("Database connection error", e);
+        }
+        return gotCustomer;
+        //Better way to return in finally with Optional. Later
+    }
+    
+    public List<Customer> getAll() throws DaoException {
+        List<Customer> set = new ArrayList<Customer>();
+        try {
+            PreparedStatement st = connectionProvider.getConnection()
+                    .prepareStatement("select * from customerapplication"
+                            + ".customer");
+            ResultSet rs = st.executeQuery();
+            set = this.convertToCustomer(rs);
+        } catch (SQLException e) {
+            throw new DaoException("Database connection error", e);
+        }
+        return set;
+    }
+    
+    public void delete(int id) throws DaoException {
+        try {
+            PreparedStatement st = connectionProvider.getConnection()
+                    .prepareStatement("delete from customerapplication"
+                            + ".customer where id=?");
+            st.setInt(1, id);
+            st.executeUpdate();
+        } catch (SQLException e) {
+            throw new DaoException("Database connection error", e);
```

Comment:
Not meaningful error message, it can be not connection issue, it can be data constraint issue, sql syntax issue... other things, Actually it is is good when exception contains all required information for finding the issue cause.